pkg/syncer/storagepool/storagepoolwatch.go

```diff
@@ -0,0 +1,127 @@
+package storagepool
+
+import (
+	"context"
+	"os"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+// SPWatcher watches for events on StoragePool CRD and for each event determines
+// if the events is caused due to external changes to StoragePool or not.
+// If yes then it triggers a reconciliation of StoragePool. It compares generation
+// number of a resource in the event with the generation number stored in
+// SPNameToGenerationNumberMap to determine external change.
+type SPWatcher struct {
+	spWatch                     watch.Interface
+	// A successful update operation to a resource instance increases its generation number by 1.
+	// So all create, delete and update operation to StoragePool should also revise gen num in this map.
+	SPNameToGenerationNumberMap map[string]int64
```

Comment:
Could we hold the expected generation number in the Intended State of the StoragePool?

Reply:
That was my first thought too when designing SP watch. But I decided not to add generation number in intended state instead use a new variable to store this information. My thoughts were along the line of: 
Information in all the fields in the intended state goes from vshpere ---> SP Contr -> Storage Pool, but information about generation number goes in opposite direction Storage Pool -> SP Contr. This may lead to, in future, special handling for this field in CRUD operation to SP which would add to software complexity. 
Also including generation number in intended state will not go along with its name and purpose as we cannot enforce the intended generation number (its updated by k8s).

Reply:
Sounds good.